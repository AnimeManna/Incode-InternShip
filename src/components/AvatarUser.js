import React, { Component } from 'react';


import PropTypes from 'prop-types';

import { withStyles } from '@material-ui/core/styles';


import {
  Avatar,
} from '@material-ui/core';

const styles = () => ({
  avatar: {
    width: 60,
    height: 60,
    border: '4px,solid,gray',
    borderRadius: '50%',
    margin: 4,
  },

});

class AvatarUser extends Component {
  constructor(props) {
    super(props);
    this.state = {
      firstLetter: '',
      lastLetter: '',
    };
    this.getLetter = this.getLetter.bind(this);
  }


  componentDidMount() {
    this.getLetter();
  }

  getLetter() {
    const { login } = this.props;
    const firstLetter = login.charAt(0);
    const numberLastLetter = login.length - 1;
    const lastLetter = login.charAt(numberLastLetter);
    this.setState({
      firstLetter,
      lastLetter,
    });
  }


  createColorFirst() {
    const { firstLetter } = this.state;
    return firstLetter.charCodeAt();
  }


  createColorLast() {
    const { lastLetter } = this.state;
    return lastLetter.charCodeAt();
  }


  render() {
    const { classes } = this.props;
    const { firstLetter, lastLetter } = this.state;
    return (
      <div>
        <Avatar className={classes.avatar} size="normal" style={{ backgroundColor: `rgb(${this.createColorFirst()},50,${this.createColorLast()})` }}>
          {firstLetter}
          {lastLetter}
        </Avatar>
      </div>
    );
  }
}

AvatarUser.propTypes = {
  classes: PropTypes.shape({
    avatar: PropTypes.string.isRequired,
  }).isRequired,
  login: PropTypes.string.isRequired,
};


export default withStyles(styles)(AvatarUser);
