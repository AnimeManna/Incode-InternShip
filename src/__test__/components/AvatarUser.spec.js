import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import AvatarUser from '../../components/AvatarUser';

describe(' <AvatarUser /> ', () => {
  const props = {
    login: 'STALIN',
  };
  describe('render() ', () => {
    test(' render the component ', () => {
      const wrapper = shallow(<AvatarUser login={props.login} />);
      const container = wrapper.dive();
      expect(toJson(container)).toMatchSnapshot();
    });
  });
});
