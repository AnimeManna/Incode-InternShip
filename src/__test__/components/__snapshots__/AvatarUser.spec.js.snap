// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` <AvatarUser />  render()   render the component  1`] = `
<div>
  <WithStyles(Avatar)
    className="AvatarUser-avatar-1"
    size="normal"
    style={
      Object {
        "backgroundColor": "rgb(83,50,78)",
      }
    }
  >
    S
    N
  </WithStyles(Avatar)>
</div>
`;
